src/Board.js

```diff
@@ -1,9 +1,10 @@
 /* Represents the board state at a point in time */
 class Board 
 {
-    constructor(adjacencyMatrix)
+    constructor(adjacencyMatrix, indexRenderer)
```

Comment:
Honestly haven't scrolled all the way through, so you might handle default values below, but would strong recommend adding a default value here. 
E.g., default to undefined, check that in the constructor and if so, grab ```IndexRenderers.getRenderer("chess")``` in here. Means one less parameter to input, and possibly fewer things down the line needing to know what IndexRenderer is.
Feel free to disagree if you think there is a strong reason to force the input.

Reply:
This is a thing I care about.